File: src/amplify/command-line/runsDirectory.ml
open Arg
open CommandLine

class c =
  object(self)
    val runsPerSubdirectory = 10000
    val root = new box "." 
    method usage_msg () = "[-runs-directory <runs directory>]"

    method argActions () =
      [("-runs-directory", String self#setRoot, "look for scripted runs in this directory (default \".\")")] 

    method setRoot r =
      root#set r

    method getRoot () = 
      root#get ()

    method format runId filename =
      let subdirectory = runId/runsPerSubdirectory in
      (root#get ())^"/"^
      (string_of_int subdirectory)^"/"^
      (string_of_int runId)^"/"^
      filename

    method cleanup () = ()
    
  end

File: src/RunsDirectory.cc
#include <sstream>
#include "RunsDirectory.h"

using namespace std;


const char *RunsDirectory::root = ".";


static const argp_option options[] = {
  {
    "runs-directory",
    'r',
    "DIR",
    0,
    "look for scripted runs in DIR (default \".\")",
    0
  },
  { 0, 0, 0, 0, 0, 0 }
};


static int
parseFlag(int key, char *arg, argp_state *)
{
  switch (key) {
  case 'r':
    RunsDirectory::root = arg;
    return 0;
  default:
    return ARGP_ERR_UNKNOWN;
  }
}


const argp RunsDirectory::argp = {
    options, parseFlag, 0, 0, 0, 0, 0
};


string
RunsDirectory::format(unsigned runId, const char filename[])
{
  ostringstream collect;
  collect << root << '/' << runId << '/' << filename;
  return collect.str();
}
